numpy/core/src/umath/loops.c.src

```diff
@@ -826,6 +832,7 @@ NPY_NO_EXPORT void
  * #TYPE = BYTE, SHORT, INT, LONG, LONGLONG#
  * #type = npy_byte, npy_short, npy_int, npy_long, npy_longlong#
  * #c    = ,,,l,ll#
+ * #div  = s32, s32, s32, s64, s64#
```

Comment:
This is incorrect on windows. The size of `npy_long` is platform-dependent.

Reply:
I went for `s64` to be on the safer side, anything else to change it to?

Reply:
I would probably add a bunch of `#define`s external to the loop, something like
```
#define libdivide_char_t NPY_CONCAT(libdivide_, NPY_BITS_OF_CHAR, _t) 
#define libdivide_char_gen NPY_CONCAT(libdivide_, NPY_BITS_OF_CHAR, _gen) 
#define libdivide_char_do NPY_CONCAT(libdivide_, NPY_BITS_OF_CHAR, _do) 
```
and repeat for the other 9 types

Reply:
Oh nice, I'll do this

Reply:
I have added some logic to auto-generate based on the above idea, please let me know if it's ok